Process finished with exit code 2

Please answer the following before submitting your issue:

Note: Please include any substantial examples (debug session output,
stacktraces, etc) as linked gists.

1. What version of Delve are you using (`dlv version`)?
2. What version of Go are you using? (`go version`)?
3. What operating system and processor architecture are you using?
4. What did you do?
5. What did you expect to see?
6. What did you see instead?

Perhaps this is a duplicate of #443 or #522, but I have a different stack trace. 
I hope, this will help to find the defect.

Please tell me, if the defect is related to IntelliJ IDEA and not to Delve.

1. delve version 0.11.0-alpha --> version of IntelliJ IDEA 2016.1 / go plugin version 0.11.1295
2. go 1.6 
3. windows / amd64
4. debugger stopped at a breakpoint (in IntelliJ IDEA)
5. go process finished with exit code 2
6. Inspect everything in IntelliJ IDEA Debugger view

stacktrace:

panic: runtime error: slice bounds out of range

goroutine 40 [running]:
github.com/derekparker/delve/proc.(*memCache).readMemory(0xc0832c4ba0, 0xffffffffffffffff, 0x40, 0x0, 0x0, 0x0, 0x0, 0x0)
	C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/proc/mem.go:23 +0x1b1
github.com/derekparker/delve/proc.readStringValue(0x3138af8, 0xc0832c4ba0, 0xffffffffffffffff, 0x51a39d, 0x0, 0x0, 0x0, 0x0)
	C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/proc/variables.go:861 +0x8b
github.com/derekparker/delve/proc.(*Variable).loadValueInternal(0xc082fe4d20, 0x2)
	C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/proc/variables.go:743 +0xda8
github.com/derekparker/delve/proc.(*Variable).loadArrayValues(0xc08207bb20, 0x1)
	C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/proc/variables.go:952 +0x2b5
github.com/derekparker/delve/proc.(*Variable).loadValueInternal(0xc08207bb20, 0x1)
	C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/proc/variables.go:747 +0x940
github.com/derekparker/delve/proc.(*Variable).loadArrayValues(0xc0833590a0, 0x0)
	C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/proc/variables.go:952 +0x2b5
github.com/derekparker/delve/proc.(*Variable).loadValueInternal(0xc0833590a0, 0x0)
	C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/proc/variables.go:747 +0x940
github.com/derekparker/delve/proc.(*Variable).loadValue(0xc0833590a0)
	C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/proc/variables.go:711 +0x31
github.com/derekparker/delve/proc.(*EvalScope).extractVariableFromEntry(0xc083316c00, 0xc0832c4960, 0x0, 0x0, 0x0)
	C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/proc/variables.go:500 +0x9f
github.com/derekparker/delve/proc.(*EvalScope).variablesByTag(0xc083316c00, 0x34, 0x0, 0x0, 0x0, 0x0, 0x0)
	C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/proc/variables.go:1491 +0x17b
github.com/derekparker/delve/proc.(*EvalScope).LocalVariables(0xc083316c00, 0x0, 0x0, 0x0, 0x0, 0x0)
	C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/proc/variables.go:531 +0x4d
github.com/derekparker/delve/service/debugger.(*Debugger).LocalVariables(0xc0820e4250, 0xffffffffffffffff, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/service/debugger/debugger.go:510 +0xc1
github.com/derekparker/delve/service/rpc.(*RPCServer).ListLocalVars(0xc0820dc8d0, 0xffffffffffffffff, 0x0, 0xc083316ba0, 0x0, 0x0)
	C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/service/rpc/server.go:288 +0x59
reflect.Value.call(0x985140, 0xa74230, 0x13, 0xa90558, 0x4, 0xc082289ed8, 0x3, 0x3, 0x0, 0x0, ...)
	c:/go/src/reflect/value.go:432 +0x1211
reflect.Value.Call(0x985140, 0xa74230, 0x13, 0xc082289ed8, 0x3, 0x3, 0x0, 0x0, 0x0)
	c:/go/src/reflect/value.go:300 +0xb8
net/rpc.(*service).call(0xc0832bd080, 0xc0832bd040, 0xc0823246e0, 0xc082328880, 0xc082ec98c0, 0x9dd1e0, 0xc082324fe0, 0xd9, 0x8ad680, 0xc083316ba0, ...)
	c:/go/src/net/rpc/server.go:383 +0x1c8
created by net/rpc.(*Server).ServeCodec
	c:/go/src/net/rpc/server.go:477 +0x4b3

goroutine 1 [chan receive]:
github.com/derekparker/delve/cmd/dlv/cmds.execute(0x0, 0xc082007e80, 0x1, 0x4, 0xc0820240c8, 0x0)
	C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/cmd/dlv/cmds/commands.go:364 +0x5e3
github.com/derekparker/delve/cmd/dlv/cmds.New.func4(0xc082074e00, 0xc082007e80, 0x1, 0x4)
	C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/cmd/dlv/cmds/commands.go:126 +0x51
github.com/derekparker/delve/vendor/github.com/spf13/cobra.(*Command).execute(0xc082074e00, 0xc082007e40, 0x4, 0x4, 0x0, 0x0)
	C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/vendor/github.com/spf13/cobra/command.go:569 +0x870
github.com/derekparker/delve/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc082074600, 0xc082074e00, 0x0, 0x0)
	C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/vendor/github.com/spf13/cobra/command.go:656 +0x572
github.com/derekparker/delve/vendor/github.com/spf13/cobra.(*Command).Execute(0xc082074600, 0x0, 0x0)
	C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/vendor/github.com/spf13/cobra/command.go:615 +0x34
main.main()
	C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/cmd/dlv/main.go:13 +0x51

goroutine 5 [syscall]:
os/signal.loop()
	c:/go/src/os/signal/signal_unix.go:22 +0x1f
created by os/signal.init.1
	c:/go/src/os/signal/signal_unix.go:28 +0x3e

goroutine 7 [chan receive, locked to thread]:
github.com/derekparker/delve/proc.(*Process).handlePtraceFuncs(0xc08207a460)
	C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/proc/proc.go:775 +0xd8
created by github.com/derekparker/delve/proc.New
	C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/proc/proc.go:77 +0x1db

goroutine 34 [IO wait]:
net.runtime_pollWait(0x1275aa8, 0x72, 0xc0823250a0)
	c:/go/src/runtime/netpoll.go:157 +0x67
net.(*pollDesc).Wait(0xc08201bdf0, 0x72, 0x0, 0x0)
	c:/go/src/net/fd_poll_runtime.go:73 +0x41
net.(*ioSrv).ExecIO(0xc0820240e0, 0xc08201bce0, 0xa93e40, 0x7, 0xb9e350, 0xc082324ff0, 0x0, 0x0)
	c:/go/src/net/fd_windows.go:182 +0x177
net.(*netFD).Read(0xc08201bc80, 0xc082074400, 0x200, 0x200, 0x0, 0x0, 0x0)
	c:/go/src/net/fd_windows.go:463 +0x17e
net.(*conn).Read(0xc0823e6010, 0xc082074400, 0x200, 0x200, 0x0, 0x0, 0x0)
	c:/go/src/net/net.go:172 +0xeb
encoding/json.(*Decoder).refill(0xc0820341a0, 0x0, 0x0)
	c:/go/src/encoding/json/stream.go:152 +0x28e
encoding/json.(*Decoder).readValue(0xc0820341a0, 0x1, 0x0, 0x0)
	c:/go/src/encoding/json/stream.go:128 +0x422
encoding/json.(*Decoder).Decode(0xc0820341a0, 0x9a6960, 0xc08226e7a0, 0x0, 0x0)
	c:/go/src/encoding/json/stream.go:57 +0x160
net/rpc/jsonrpc.(*serverCodec).ReadRequestHeader(0xc08226e780, 0xc082ec9760, 0x0, 0x0)
	c:/go/src/net/rpc/jsonrpc/server.go:66 +0x83
net/rpc.(*Server).readRequestHeader(0xc0832bd040, 0x309a750, 0xc08226e780, 0x0, 0x0, 0xc082ec9760, 0xc08228de00, 0x0, 0x0)
	c:/go/src/net/rpc/server.go:576 +0x97
net/rpc.(*Server).readRequest(0xc0832bd040, 0x309a750, 0xc08226e780, 0xc0832bd040, 0xc0823246e0, 0xc082328880, 0x0, 0x0, 0x0, 0x0, ...)
	c:/go/src/net/rpc/server.go:543 +0x92
net/rpc.(*Server).ServeCodec(0xc0832bd040, 0x309a750, 0xc08226e780)
	c:/go/src/net/rpc/server.go:462 +0x93
created by github.com/derekparker/delve/service/rpc.(*ServerImpl).Run.func1
	C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/service/rpc/server.go:90 +0x1b2

Process finished with exit code 2

--- a/proc/mem.go
+++ b/proc/mem.go
@@ -14,7 +14,7 @@ type memCache struct {
 }
 
 func (m *memCache) contains(addr uintptr, size int) bool {
-	return addr >= m.cacheAddr && (addr+uintptr(size)) <= (m.cacheAddr+uintptr(len(m.cache)))
+	return addr >= m.cacheAddr && addr <= (m.cacheAddr+uintptr(len(m.cache) - size))
 }
 
 func (m *memCache) readMemory(addr uintptr, size int) (data []byte, err error) {

--- a/proc/proc_test.go
+++ b/proc/proc_test.go
@@ -1741,3 +1741,12 @@ func TestIssue462(t *testing.T) {
 		assertNoError(err, t, "Stacktrace()")
 	})
 }
+
+func TestIssue554(t *testing.T) {
+	// unsigned integer overflow in proc.(*memCache).contains was
+	// causing it to always return true for address 0xffffffffffffffff
+	mem := memCache{0x20, make([]byte, 100), nil}
+	if mem.contains(0xffffffffffffffff, 40) {
+		t.Fatalf("should be false")
+	}
+}

